pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -195,15 +196,33 @@ mod sync_queue {
     /// or two (download and upload, if both were found in the queue during batch construction).
     pub async fn next_task_batch(
         receiver: &mut UnboundedReceiver<(ZTenantTimelineId, SyncTask)>,
-        mut max_batch_size: usize,
-    ) -> HashMap<ZTenantTimelineId, SyncTask> {
-        if max_batch_size == 0 {
-            return HashMap::new();
+        max_timelines_to_sync: usize,
```

Comment:
Maybe it is not a good idea, but NonZeroUsize should be able to eliminate zero case handled below

Reply:
I think this is a great idea actually, fixed.